store/retry.go

```diff
@@ -70,6 +71,9 @@ func shouldRetryError(attempt *retry.Attempt, err error) bool {
 	if !attempt.More() {
 		return false
 	}
+	if urlErr, ok := err.(*url.Error); ok {
+		return urlErr.Err == io.ErrUnexpectedEOF || urlErr.Err == io.EOF
```

Comment:
Instead of repeating the check here, it'd be better to just unwrap and let the follow up code run
```
if urlErr, ok := err.(*url.Error); ok {
        err = urlErr
}
```